File: apps/api/AssetManagement.Api/DTOs/SearchDtos.cs
namespace AssetManagement.Api.DTOs;

public record SearchResultItem(string Id, string Name, string? Subtitle);

public record SearchResponse(
    List<SearchResultItem> Assets,
    List<SearchResultItem> Certificates,
    List<SearchResultItem> Applications,
    List<SearchResultItem> People,
    List<SearchResultItem> Locations);

File: apps/api-kt/src/main/kotlin/com/assetmanagement/api/dto/SearchDtos.kt
package com.assetmanagement.api.dto

import java.util.*

data class SearchResultItem(
    val id: UUID,
    val name: String,
    val subtitle: String? = null,
    val extra: String? = null
)

data class SearchCounts(
    val assets: Int,
    val certificates: Int,
    val applications: Int,
    val people: Int,
    val locations: Int
)

data class SearchResponse(
    val assets: List<SearchResultItem>,
    val certificates: List<SearchResultItem>,
    val applications: List<SearchResultItem>,
    val people: List<SearchResultItem>,
    val locations: List<SearchResultItem>,
    val counts: SearchCounts = SearchCounts(0, 0, 0, 0, 0)
)
